readthedocs/ssh/signals.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+"""Signals to use private/public keys before/after events."""
+
+from __future__ import division, print_function, unicode_literals
+
+import logging
+import glob
+import os
+import re
+import signal
+import subprocess
+
+from django.dispatch import receiver
+
+from readthedocs.doc_builder.base import restoring_chdir
+from readthedocs.projects.signals import after_vcs, before_vcs
+from readthedocs.restapi.client import api
+
+log = logging.getLogger(__name__)
+
+
+@receiver(before_vcs)
+@restoring_chdir
+def setup_ssh_agent(sender, **kwargs):
+    """
+    Start SSH Agent for this process.
+
+    Hit the API endpoint to retrieve all the SSH keys for this project and load
+    them temporarily into the SSH agent using ``ssh-add`` command.
+    """
+    log.info('Setting up SSH Agent')
+    version = sender
+    key_path = os.path.join(version.project.doc_path, 'checkouts')
```

Comment:
This doesn't look new, but we should not be doing this. We should be using a temp dir via `tmpdir`, so that we can ensure the keys are completely off the server after being added to the agent. Inside `checkouts`, the keys will persist across builds. The key here is to keep the private keys out of the docker container and shared storage.